airbyte-workers/src/main/java/io/airbyte/workers/process/KubePodProcess.java

```diff
@@ -418,6 +423,13 @@ private boolean isTerminal(Pod pod) {
 
   private int getReturnCode(Pod pod) {
     Pod refreshedPod = client.pods().inNamespace(pod.getMetadata().getNamespace()).withName(pod.getMetadata().getName()).get();
+    if (refreshedPod == null) {
```

Comment:
I don't think this is safe. If the pod was never created for some reason we shouldn't allow this to return `0`.

Reply:
what if we took the approach where we keep a class variable `terminated` that is set to true if either `waitFor` or `destroy` are called? we can return 0 if `terminated` is true and go back to throwing an exception if the pod is null.

the alternative is keeping terminated pods around, which is icky.